src/movies.js

```diff
@@ -1,25 +1,114 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  let directorsName = moviesArray.map((i) => {
+    return i.director;
+  });
+  return directorsName;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  let spielbergMovies = moviesArray.filter((el) => {
+    if (el.director === "Steven Spielberg" && el.genre.includes("Drama")) {
+      return el;
+    }
+  });
+  return spielbergMovies.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  if (moviesArray.length === 0) {
+    return 0;
+  }
+  let mediumScore = moviesArray.reduce((acc, film) => {
+    if (film.score) {
+      return acc + film.score;
+    } else {
+      return acc;
+    }
+  }, 0); //always set the initial value to start the acc
+  return Math.round((mediumScore * 100) / moviesArray.length) / 100;
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+  let scoresDramaArray = [];
+  let avg = 0;
+
+  scoresDramaArray = moviesArray.filter((movie) => {
+    return movie.genre.includes("Drama");
+  });
+
+  scoresDramaArray = scoresDramaArray.map((movie) => {
+    return movie.score;
+  });
+
+  let total = scoresDramaArray.reduce((acc, film) => {
```

Comment:
Previously, you wrote a function that calculates the average scores, right? So, here, you should reuse it. Instead of writing again the logic to calculate the average, you can call the `scoresAverage` function.
`return scoresAverage(scoresDramaArray)`.

Reply:
thank you!